src/main/webapp/script.js

```diff
@@ -12,9 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
 var arrayOfProducts = Array();
 
 async function fetchDataAsync() {
     const response = await fetch('test.html');
     arrayOfProducts = await response.json();
+}
+
+function filtering(){
```

Comment:
Looks like these functions you added are not used anywhere yet (or the entire js file is not used anywhere yet)? It would be nice to not having "leave alone" code in your codebase because we have 0 clue on if the code is actually working or not. 

I would suggest you to remove these code from this PR and put it in a future PR where these code are actually being used.